jme3-plugins/src/fbx/java/com/jme3/scene/plugins/fbx/objects/FbxMesh.java

```diff
@@ -332,8 +332,8 @@ private List<Geometry> createGeometries() throws IOException {
             indexMap = new ArrayList<>(vCount);
             reverseVertexMap = new ArrayList<>(vCount);
             for(int i = 0; i < vCount; ++i) {
-                vertexMap.set(i, i);
-                indexMap.set(i, i);
+                vertexMap.add(i, i);
+                indexMap.add(i, i);
```

Comment:
The set method does replace one element of the collection with another. However, if an element with the given index does not exist, the method will generate an IndexOutOfBoundExeption error. In the above code, the collection is created before the loop, so it will always be empty.

Reply:
technically add(index,v) is still wrong, the only reason it works is that it is executed in sequence from 0 to n, so i suggest to change it to 
```java
vertexMap.add(i);
indexMap.add(i);
```
without the index

Reply:
the vCount initializes the capacity, that is indeed the initial size of the internal array, however set will still fail because size() is not == capacity in arraylist . Kinda weird, but that's how it is, it would need a `for(0..capacity) vertexMap.add(null)` for set to work as expected